pennylane/devices/default_qubit.py

```diff
@@ -189,13 +190,12 @@ def _apply_state_vector(self, state, device_wires):
         # translate to wire labels used by device
         device_wires = self.map_wires(device_wires)
 
-        state = self._asarray(state, dtype=self.C_DTYPE)
         n_state_vector = state.shape[0]
 
         if state.ndim != 1 or n_state_vector != 2 ** len(device_wires):
             raise ValueError("State vector must be of length 2**wires.")
 
-        if not np.allclose(np.linalg.norm(state, ord=2), 1.0, atol=tolerance):
+        if not math.isclose(np.linalg.norm(state, ord=2), 1.0, abs_tol=tolerance):
```

Comment:
Do we need to make `np.linalg.norm` a static method? E.g. `self._norm()`? Then this would be overwritten e.g. in the `default.qubit.tf` device. Although, I don't think it matters for TF :thinking:

Reply:
This should be fine, since this isn't part of the computational graph